clouddriver-ecs/src/test/groovy/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperationSpec.groovy

```diff
@@ -724,4 +724,540 @@ class CreateServerGroupAtomicOperationSpec extends CommonAtomicOperation {
+      .withScalableTargets(new ScalableTarget()
+      .withResourceId("service/test-cluster/${serviceName}-v007")
+      .withMinCapacity(2)
+      .withMaxCapacity(4))
+
+    1 * operation.ecsCloudMetricService.copyScalingPolicies(
+      "Test",
+      "us-west-1",
+      "${serviceName}-v008",
+      "service/test-cluster/${serviceName}-v008",
+      "test",
+      "us-west-1",
+      "${serviceName}-v007",
+      "service/test-cluster/${serviceName}-v007",
+      "test-cluster"
+    )
+  }
+
+  def 'create a service with different TargetGroupMappings and deprecated target group properties'() {
+    given:
+    def source = new CreateServerGroupDescription.Source()
+    source.account = "test"
+    source.region = "us-west-1"
+    source.asgName = "${serviceName}-v007"
+    source.useSourceCapacity = true
+
+    def targetGroupProperty = new CreateServerGroupDescription.TargetGroupProperties(
+      containerPort: 80,
+      containerName: 'v009',
+      targetGroup: 'target-group-arn'
+    )
+
+    def placementConstraint = new PlacementConstraint(type: 'memberOf', expression: 'attribute:ecs.instance-type =~ t2.*')
+
+    def placementStrategy = new PlacementStrategy(type: 'spread', field: 'attribute:ecs.availability-zone')
+
+    def description = new CreateServerGroupDescription(
+      credentials: TestCredential.named('Test', [:]),
+      application: applicationName,
+      stack: stack,
+      freeFormDetails: detail,
+      ecsClusterName: 'test-cluster',
+      iamRole: 'test-role',
+      containerPort: 1337,
+      targetGroup: 'target-group-arn',
+      targetGroupMappings: [targetGroupProperty],
+      portProtocol: 'tcp',
+      computeUnits: 9001,
+      tags: ['label1': 'value1', 'fruit': 'tomato'],
+      reservedMemory: 9002,
+      dockerImageAddress: 'docker-image-url',
+      capacity: new ServerGroup.Capacity(1, 1, 1),
+      availabilityZones: ['us-west-1': ['us-west-1a', 'us-west-1b', 'us-west-1c']],
+      placementStrategySequence: [placementStrategy],
+      placementConstraints: [placementConstraint],
+      source: source
+    )
+
+    def operation = new CreateServerGroupAtomicOperation(description)
+
+    operation.amazonClientProvider = amazonClientProvider
+    operation.ecsCloudMetricService = Mock(EcsCloudMetricService)
+    operation.iamPolicyReader = iamPolicyReader
+    operation.accountCredentialsProvider = accountCredentialsProvider
+    operation.containerInformationService = containerInformationService
+
+    amazonClientProvider.getAmazonEcs(_, _, _) >> ecs
+    amazonClientProvider.getAmazonIdentityManagement(_, _, _) >> iamClient
+    amazonClientProvider.getAmazonElasticLoadBalancingV2(_, _, _) >> loadBalancingV2
+    amazonClientProvider.getAmazonApplicationAutoScaling(_, _, _) >> autoScalingClient
+    containerInformationService.getClusterName(_, _, _) >> 'cluster-name'
+    accountCredentialsProvider.getCredentials(_) >> creds
```

Comment:
Do these need to be repeated in every test?  Can they be moved up to a common setup?

Reply:
Addressed